tools/lib/html_branches.py

```diff
@@ -163,3 +166,15 @@ def html_tag_tree(text):
             stack.pop()
 
     return top_level
+
+def check_Dupid(templates):
+    # type: (List[str]) -> None
+    master_Idrecord = {} # type: (Dict[str,List[str]])
+    for lnk in templates:
+        text = open(lnk).read()
+        html_branches(text,fn=lnk,master_Idrecord=master_Idrecord)
```

Comment:
Here is where I'm thinking we just take the output from html_branches and run through it to collect the list of ids and update our dictionary.